components/director/internal/domain/bundlereferences/repository.go

```diff
@@ -0,0 +1,179 @@
+	updatableColumnsWithoutBundleID = []string{"api_def_id", "event_def_id", "api_def_url"}
+)
+
+//go:generate mockery --name=BundleReferenceConverter --output=automock --outpkg=automock --case=underscore
+type BundleReferenceConverter interface {
+	ToEntity(in model.BundleReference) Entity
+	FromEntity(in Entity) (model.BundleReference, error)
+}
+
+type repository struct {
+	creator         repo.Creator
+	pageableQuerier repo.PageableQuerier
+	lister          repo.Lister
+	getter          repo.SingleGetter
+	deleter         repo.Deleter
+	updater         repo.Updater
+	existQuerier    repo.ExistQuerier
+	conv            BundleReferenceConverter
+}
+
+func NewRepository(conv BundleReferenceConverter) *repository {
+	return &repository{
+		creator:         repo.NewCreator(resource.BundleReference, BundleReferenceTable, bundleReferencesColumns),
+		pageableQuerier: repo.NewPageableQuerier(resource.BundleReference, BundleReferenceTable, tenantColumn, bundleReferencesColumns),
+		lister:          repo.NewLister(resource.BundleReference, BundleReferenceTable, tenantColumn, bundleReferencesColumns),
+		getter:          repo.NewSingleGetter(resource.BundleReference, BundleReferenceTable, tenantColumn, bundleReferencesColumns),
+		deleter:         repo.NewDeleter(resource.BundleReference, BundleReferenceTable, tenantColumn),
+		updater:         repo.NewUpdater(resource.BundleReference, BundleReferenceTable, updatableColumns, tenantColumn, []string{}),
+		existQuerier:    repo.NewExistQuerier(resource.BundleReference, BundleReferenceTable, tenantColumn),
+		conv:            conv,
+	}
+}
+
+type BundleReferencesCollection []Entity
+
+func (r BundleReferencesCollection) Len() int {
+	return len(r)
+}
+
+func (r *repository) GetByID(ctx context.Context, objectType model.BundleReferenceObjectType, tenantID string, objectID, bundleID *string) (*model.BundleReference, error) {
+	fieldName, err := r.referenceObjectFieldName(objectType)
+	if err != nil {
+		return nil, err
+	}
+
+	var bundleReferenceEntity Entity
+	var conditions repo.Conditions
+
+	if bundleID == nil {
+		conditions = repo.Conditions{repo.NewEqualCondition(fieldName, objectID)}
+	} else {
+		conditions = repo.Conditions{
+			repo.NewEqualCondition(fieldName, objectID),
+			repo.NewEqualCondition(bundleIDColumn, bundleID),
+		}
+	}
+
+	err = r.getter.Get(ctx, tenantID, conditions, repo.NoOrderBy, &bundleReferenceEntity)
+	if err != nil {
+		return nil, err
+	}
+
+	bundleReferenceModel, err := r.conv.FromEntity(bundleReferenceEntity)
+	if err != nil {
+		return nil, err
+	}
+
+	return &bundleReferenceModel, nil
+}
+
+func (r *repository) GetBundleIDsForObject(ctx context.Context, tenantID string, objectType model.BundleReferenceObjectType, objectID *string) (ids []string, err error) {
+	fieldName, err := r.referenceObjectFieldName(objectType)
+	if err != nil {
+		return nil, err
+	}
+
+	var objectBundleIDs IDs
+
+	r.lister.SetSelectedColumns([]string{bundleIDColumn})
```

Comment:
as the lister is one for the repository and the repository is shared between resolver requests it might happen that multiple requests use this method and try to change a property of the object which will lead to concurrency problem.
Perhaps instantiating a new instance of the lister on change will have to be done in order not to use mutex which will be slower IMHO.